packages/astro/src/cli/check/index.ts

```diff
@@ -1,386 +1,39 @@
-			logging: this.#logging,
-			fs: this.#fs,
-		});
-		// early exit on sync failure
-		if (processExit === 1) return processExit;
-
-		let spinner = ora(
-			` Getting diagnostics for Astro files in ${fileURLToPath(this.#settings.config.root)}…`
-		).start();
-
-		if (openDocuments) {
-			await this.#openDocuments();
-		}
-
-		let diagnostics = await this.#diagnosticsChecker.getDiagnostics();
-
-		spinner.succeed();
-
-		let brokenDownDiagnostics = this.#breakDownDiagnostics(diagnostics);
-		this.#logDiagnosticsSeverity(brokenDownDiagnostics);
-		return brokenDownDiagnostics.errors > 0
-			? CheckResult.ExitWithError
-			: CheckResult.ExitWithSuccess;
-	}
-
-	#checkForDiagnostics() {
-		clearTimeout(this.#updateDiagnostics);
-		// @ematipico: I am not sure of `setTimeout`. I would rather use a debounce but let's see if this works.
-		// Inspiration from `svelte-check`.
-		this.#updateDiagnostics = setTimeout(async () => await this.#checkAllFiles(false), 500);
-	}
-
-	/**
-	 * This function is responsible to attach events to the server watcher
-	 */
-	async #watch() {
-		const { default: chokidar } = await import('chokidar');
-		this.#watcher = chokidar.watch(
-			join(fileURLToPath(this.#settings.config.root), ASTRO_GLOB_PATTERN),
-			{
-				ignored: ['**/node_modules/**'],
-				ignoreInitial: true,
-			}
-		);
-
-		this.#watcher.on('add', (file) => {
-			this.#addDocument(file);
-			this.#filesCount += 1;
-			this.#checkForDiagnostics();
-		});
-		this.#watcher.on('change', (file) => {
-			this.#addDocument(file);
-			this.#checkForDiagnostics();
-		});
-		this.#watcher.on('unlink', (file) => {
-			this.#diagnosticsChecker.removeDocument(file);
-			this.#filesCount -= 1;
-			this.#checkForDiagnostics();
-		});
-	}
-
-	/**
-	 * Add a document to the diagnostics checker
-	 * @param filePath Path to the file
-	 */
-	#addDocument(filePath: string) {
-		const text = fs.readFileSync(filePath, 'utf-8');
-		this.#diagnosticsChecker.upsertDocument({
-			uri: pathToFileURL(filePath).toString(),
-			text,
-		});
-	}
+		getPackageOpts,
+		['typescript']
+	);
+	const typescript = await getPackage('typescript', logging, getPackageOpts);
 
-	/**
-	 * Logs the result of the various diagnostics
-	 *
-	 * @param result Result emitted by AstroChecker.#breakDownDiagnostics
-	 */
-	#logDiagnosticsSeverity(result: Readonly<DiagnosticResult>) {
-		info(
-			this.#logging,
-			'diagnostics',
-			[
-				bold(`Result (${this.#filesCount} file${this.#filesCount === 1 ? '' : 's'}): `),
-				bold(red(`${result.errors} ${result.errors === 1 ? 'error' : 'errors'}`)),
-				bold(yellow(`${result.warnings} ${result.warnings === 1 ? 'warning' : 'warnings'}`)),
-				dim(`${result.hints} ${result.hints === 1 ? 'hint' : 'hints'}\n`),
-			].join(`\n${dim('-')} `)
+	if (!checkPackage || !typescript) {
+		error(
+			logging,
+			'check',
+			'The `@astrojs/check` and `typescript` packages are required for this command to work. Please install them into your project and try again.'
```

Comment:
nit: I think the message is misleading. The `getPackage` method actually **tries** to install the package, so if it's `undefined` the cause could also be that, right?

Reply:
Hmm, I don't think it's necessarily misleading, if the install fails, you'll get an error message from it before this message appear. I could add a "manually" in there though